lib/AST/ASTContext.cpp

```diff
@@ -3173,8 +3180,11 @@ PackExpansionType::PackExpansionType(Type patternType, Type countType,
                                      const ASTContext *canCtx)
   : TypeBase(TypeKind::PackExpansion, canCtx, properties),
     patternType(patternType), countType(countType) {
+
+  // TODO: it would be nice if the solver didn't stick PlaceholderTypes in here.
```

Comment:
To do that, don't set `TVO_CanBindToHole` on the type variable that represents the count.